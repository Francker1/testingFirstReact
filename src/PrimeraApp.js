
import PropTypes from "prop-types";

//functional compontents

const PrimeraApp = ( { nombre, edad, despedida, defol } ) => {

    return (
        <>
            <h1>Hola Mundo</h1>
            <p>Mi nombre es { nombre }</p>
            <p>Tengo { edad || "X" } a√±os</p>
            <p>{ despedida }</p>
            <p>{ defol }</p>
        </>
    );

}

//defincion de proptypes
PrimeraApp.propTypes = {
    nombre: PropTypes.string.isRequired,
    edad: PropTypes.number,
    despedida: PropTypes.string.isRequired,
}

//definicion de propiedades default, si no son informadas, de esta forma si se ven en Componentes F12
PrimeraApp.defaultProps = {
    defol: "Soy default",
}

export default PrimeraApp;