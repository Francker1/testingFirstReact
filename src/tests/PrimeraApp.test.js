
import React from 'react';
// import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import PrimeraApp from "../PrimeraApp";



describe('Pruebas en <PrimeraApp />', () => {
    

    // test("debe de mostrar el mensaje 'Mi nombre es Ítalo'", () => {
      // const nombre = "Ítalo";
      // const despedida = "Adiós."
      // const wrapper = render( <PrimeraApp nombre={nombre} despedida={despedida} /> );
      // expect(wrapper.getByText("Mi nombre es Ítalo")).toBeInTheDocument();
    // });
    
    test("debe mostrar <PrimeraApp /> correctamente", () => {
      
      const nombre = "Ítalo";
      const despedida = "Adiós.";

      const wrapper = shallow(
        <PrimeraApp nombre={nombre} despedida={despedida} />
      );

        expect(wrapper).toMatchSnapshot();
    });


    test("debe mostrar la despedida enviada por props", () => {

        const nombre = "Ítalo";
        const despedida = "Adiós.";

        const wrapper = shallow(
          <PrimeraApp nombre={nombre} despedida={despedida} />
        );

        const textoDespedida = wrapper.find('h3').text();
        
        expect(textoDespedida).toBe(despedida);
    });
});
