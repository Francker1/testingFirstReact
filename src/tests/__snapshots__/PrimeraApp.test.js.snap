// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola Mundo!!
  </h1>
  <p>
    Mi nombre es 
    Ítalo
  </p>
  <p>
    Tengo 
    X
     años
  </p>
  <h3>
    Adiós.
  </h3>
  <p>
    Soy default
  </p>
</Fragment>
`;
